src/UI/Art/SpriteArtist/Sprite.java

```diff
@@ -111,9 +111,7 @@ public void addFrame(BufferedImage sprite)
 
   public BufferedImage getFrame(int index)
   {
-    // Normalize the index if needed.
-    for(; index >= spriteImages.size() || index < 0; index += spriteImages.size() * ((index < 0)? 1:-1));
-    return spriteImages.get(index);
+    return spriteImages.get(Math.abs(index) % spriteImages.size());
```

Comment:
Good point. I should fix that.
This loop caused horrible lag on my machine because the `index` coming in was very large (due to the change in how we calculate animation indices)

(This would have happened anyway previously, it would have just taken a while to get the index that big)